Tests/Add-ZipArchiveEntry.Tests.ps1

```diff
@@ -245,29 +229,59 @@ Describe 'Add-ZipArchiveEntry.when adding archive root' {
 Describe 'Add-ZipArchiveEntry.when passing path instead of file objects' {
     Init
     GivenFile 'one.cs','two.cs'
-    WhenAddingFiles '*.cs' -NoPipeline
+    WhenAddingFiles '*.cs'
     ThenArchiveContains 'one.cs','two.cs'
 }
 
-Describe 'Add-ZipArchiveEntry.when no base path' {
+Describe 'Add-ZipArchiveEntry.when changing name' {
     Init
-    GivenFile 'one.cs','two.cs'
-    WhenAddingFiles '*.cs' -NoBasePath
-    $rootPath = $TestDrive.FullName -replace '^[^:]:\\',''
-    ThenArchiveContains ('one.cs','two.cs' | ForEach-Object { '{0}\{1}' -f $rootPath,$_ })
+    GivenFile 'one.cs'
+    WhenAddingFiles 'one.cs' -WithName 'cs.one'
+    ThenArchiveContains 'cs.one'
+    ThenArchiveNotContains 'one.cs'
 }
 
 Describe 'Add-ZipArchiveEntry.when passing a directory' {
     Init
-    GivenFile 'dir1\one.cs','dir1\two.cs'
-    WhenAddingFiles (Join-Path -Path $TestDrive.FullName -ChildPath 'dir1') -NoPipeline
-    ThenArchiveContains 'dir1\one.cs','dir1\two.cs'
+    GivenFile 'dir1\one.cs','dir1\two.cs', 'dir1\three\four.cs'
+    WhenAddingFiles 'dir1'
+    ThenArchiveContains 'dir1\one.cs','dir1\two.cs','dir1\three\four.cs'
 }
 
-Describe 'Add-ZipArchiveEntry.when giving an item a new root name' {
+Describe 'Add-ZipArchiveEntry.when customizing a directory name' {
+    Init
+    GivenFile 'dir1\one.cs','dir1\two.cs', 'dir1\three\four.cs'
+    WhenAddingFiles 'dir1' -WithName '1dir'
+    ThenArchiveContains '1dir\one.cs','1dir\two.cs','1dir\three\four.cs'
+    ThenArchiveNotContains 'dir1\one.cs','dir1\two.cs','dir1\three\four.cs'
+}
+
+Describe 'Add-ZipArchiveEntry.when passing a directory with a custom base path' {
+    Init
+    GivenFile 'dir1\one.cs','dir1\two.cs', 'dir1\three\four.cs'
+    WhenAddingFiles 'dir1'

```

Comment:
Describe says `when passing a directory with a custom base path` but no base path parameter is given.